packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -187,3 +198,82 @@ export abstract class LLMStrategy<I extends OptimisticOracleRequest, R extends O
     return this.results;
   }
 }
+
+/**
+ * Enum for options available for handling strategy results.
+ */
+export enum StrategyHandlerOptions {
+  Dispute = "Dispute",
+  Propose = "Propose",
+  Log = "Log",
+  Backtest = "Backtest",
+}
+
+/**
+ * Class for handling the results of an LLMStrategy.
+ * This can be used to execute further actions based on the strategy results, such as disputing or proposing new requests, logging, or backtesting.
+ * @template R The type of the results from the LLMStrategy.
+ */
+export abstract class OptimisticOracleStrategyHandler<R extends OptimisticOracleRequest> {
+  protected strategyResults: R[];
+
+  /**
+   * Creates a new OptimisticOracleStrategyHandler.
+   * @param strategyResults The results from an LLMStrategy to be handled.
+   */
+  constructor(strategyResults: R[]) {
+    this.strategyResults = strategyResults;
+  }
```

Comment:
Just OOC, why did you decide to store these in the constructor instead of passing them into the process() function call?